hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/SparkRDDWriteClient.java

```diff
@@ -299,6 +302,7 @@ protected void completeCompaction(HoodieCommitMetadata metadata, JavaRDD<WriteSt
     this.context.setJobStatus(this.getClass().getSimpleName(), "Collect compaction write status and commit compaction");
     List<HoodieWriteStat> writeStats = writeStatuses.map(WriteStatus::getStat).collect();
     finalizeWrite(table, compactionCommitTime, writeStats);
+    table.getMetadataWriter().ifPresent(w -> w.update(metadata, compactionCommitTime));
```

Comment:
nts: audit all paths that commit to timeline and ensure this is done everywhere. this may also be an opportunity to streamline such code occurrences.